blender/LilySurfaceScrapper/Scrappers/TexturesOneScrapper.py

```diff
@@ -24,21 +38,33 @@ def findSource(cls, url):
     @classmethod
     def canHandleUrl(cls, url):
         """Return true if the URL can be scrapped by this scrapper."""
-        if "textures.one/go/?id=" in url: # this is superior to url.startswith(), because it can deal with leaving out "https://" or "www."
+        if "textures.one/go/?id=" in url:
             source_url = cls.findSource(url)
             if source_url is not None:
                 # Look for a scrapper that can scrape the source page
                 for S in ScrappersManager.getScrappersList():
                     if S.canHandleUrl(source_url):
+                        cls.source_url = source_url
                         cls.source_scrapper_type = S
-                        cls.source_scrapper = S("") # I'd love to create this in this classes __init__(), but it gave me headache with scope problems.
                         cls.scrapped_type = cls.source_scrapper_type.scrapped_type # This works
```

Comment:
What you are achieving with these class attributes (`cls.something`) already is a caching mechanism. The class caches the information that to the given URL is associated this or that scrapper class, and the information is retrieved in the constructor. Only it is a cache with a capacity of exactly one value, meaning that it overflows as soon as `canHandleUrl` is called twice (in other terms, the cached value of `cls.source_scrapper_type` and the likes is overwritten).

It is not a big deal, you'd say, because `__init__` is always called right afterward with the same URL. I agree. In the way the class is used by the reminding of the code, it won't break. But my concern is that nothing ensures this, i.e. nothing warns and reminds future developpers (including future ourselves in two weeks) that it has to remain so. This is why I would find it cleaner to clearly state, in the naming and so on, that there is a caching mechanism involved.

So I would suggest a pattern a bit like this:

```python
_url_cache = {}
#...
# in canHandleUrl(), after S.canHandleUrl(source_url):
cls._url_cache[url] = (source_url, S)
# ...
# in fetchVariantList():
if url not in cls._url_cache:
    return []
source_url, S = cls._url_cache[url]
self.source_scrapper = S(self.texture_root)
# ...
```